lib/types/local-time.js

```diff
@@ -29,249 +29,182 @@ const nanoSecInMillis = Long.fromNumber(1000000);
- * @class
- * @classdesc A time without a time-zone in the ISO-8601 calendar system, such as 10:30:05.
- * <p>
- *   LocalTime is an immutable date-time object that represents a time, often viewed as hour-minute-second. Time is represented to nanosecond precision. For example, the value "13:45.30.123456789" can be stored in a LocalTime.
- * </p>
- * @param {Long} totalNanoseconds Total nanoseconds since midnight.
- * @constructor
- */
-function LocalTime(totalNanoseconds) {
-    if (!(totalNanoseconds instanceof Long)) {
-        throw new Error("You must specify a Long value as totalNanoseconds");
-    }
-    if (
-        totalNanoseconds.lessThan(Long.ZERO) ||
-        totalNanoseconds.greaterThan(maxNanos)
-    ) {
-        throw new Error("Total nanoseconds out of range");
-    }
 
-    this.internal = rust.LocalTime.new(longToBigint(totalNanoseconds));
+class LocalTime {
+    /**
+     *
+     * Creates a new instance of LocalTime.
+     * @class
+     * @classdesc A time without a time-zone in the ISO-8601 calendar system, such as 10:30:05.
+     * <p>
+     *   LocalTime is an immutable date-time object that represents a time, often viewed as hour-minute-second. Time is represented to nanosecond precision. For example, the value "13:45.30.123456789" can be stored in a LocalTime.
+     * </p>
+     * @param {Long} totalNanoseconds Total nanoseconds since midnight.
+     * @constructor
+     */
+    constructor(totalNanoseconds) {
+        if (!(totalNanoseconds instanceof Long)) {
+            throw new Error(
+                "You must specify a Long value as totalNanoseconds",
+            );
+        }
+        if (
+            totalNanoseconds.lessThan(Long.ZERO) ||
+            totalNanoseconds.greaterThan(maxNanos)
+        ) {
+            throw new Error("Total nanoseconds out of range");
+        }
 
-    //tis use unit tests ...
-    this.value = bigintToLong(this.internal.value);
-    this.nanosecond = this.internal.nanosecond;
-    this.second = this.internal.second;
-    this.minute = this.internal.minute;
-    this.hour = this.internal.hour;
-}
+        this.internal = rust.LocalTime.new(longToBigint(totalNanoseconds));
 
-/**
- * Parses an string representation and returns a new LocalDate.
- * @param {String} value
- * @returns {LocalTime}
- */
-LocalTime.fromString = function (value) {
-    if (typeof value !== "string") {
-        throw new Error("Argument type invalid: " + util.inspect(value));
-    }
-    const parts = value.split(":");
-    let millis =
-        parseInt(parts[0], 10) * 3600000 + parseInt(parts[1], 10) * 60000;
-    let nanos;
-    if (parts.length === 3) {
-        const secParts = parts[2].split(".");
-        millis += parseInt(secParts[0], 10) * 1000;
-        if (secParts.length === 2) {
-            nanos = secParts[1];
-            //add zeros at the end
-            nanos = nanos + utils.stringRepeat("0", 9 - nanos.length);
-        }
+        //this use unit tests ...
```

Comment:
What is the subject here? I assume it's unit tests who use this code, not the other way round; let's not abuse English syntax :slightly_smiling_face: 
`this is used by unit tests` or `unit tests use this`